base/server/src/main/java/org/dogtagpki/server/rest/JSONSerializer.java

```diff
@@ -0,0 +1,19 @@
+package org.dogtagpki.server.rest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+/**
+ * Implement this interface to provide default methods to serialize an object to/from JSON
+ */
+public interface JSONSerializer {
+
+    default String toJSON() throws Exception {
+        return new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).writeValueAsString(this);
+    }
+
+    static Object fromJSON(String json, Class<?> clazz) throws Exception {
+        return new ObjectMapper().readValue(json, clazz);
+    }
```

Comment:
Hmm.. will `fromJSON()` invocations look like this?
```
ACMEDatabaseConfig config = (ACMEDatabaseConfig) ACMEDatabaseConfig.fromJSON(
    json, ACMEDatabaseConfig.class);
```
Is there a way to simplify it, maybe using generics?

Reply:
Yes. It's a bit verbose, but I would argue it would still be an improvement over duplicating two methods ~100 times.

I'm looking at trying to simplify it.

Reply:
@edewata I can simplify it, but not much, the best I can do is make this:
```
ACMEDatabaseConfig config = (ACMEDatabaseConfig) JSONSerializer.fromJSON(
    json, ACMEDatabaseConfig.class);
```
into this:
```
var config = (ACMEDatabaseConfig) JSONSerializer.fromJSON(
    json, ACMEDatabaseConfig.class);
```
Not an amazing saving, but does drop one repetition of the class name. The problem boils down to "I want the return type of my method to match the return type of one of the parameters", which would allow you to not cast which but I'm not sure is possible.

Reply:
Would something like this work?
https://github.com/dogtagpki/pki/blob/master/base/common/src/main/java/com/netscape/certsrv/client/PKIClient.java#L198

If not that's fine, but could you add some unit tests so we can see the actual usage?

Reply:
It does, thanks! The tests added for the JSON mappers will all be modified to point to this interface, so it is well covered already I think - there are just no tests for any of these ACME classes (yet)!